autoload/SpaceVim/plugins/runner.vim

```diff
@@ -392,16 +395,63 @@ function! SpaceVim#plugins#runner#run_task(task) abort
     if !empty(opts) && has_key(opts, 'env') && !empty(opts.env)
       call extend(opt, {'env' : opts.env})
     endif
+    let problemMatcher = get(a:task, 'problemMatcher', {})
     if isBackground
-      call s:run_backgroud(cmd, opt)
+      call s:run_backgroud(cmd, opt, problemMatcher)
     else
-      call SpaceVim#plugins#runner#open(cmd, opt) 
+      call SpaceVim#plugins#runner#open(cmd, opt, problemMatcher) 
     endif
   endif
 endfunction
 
+function! s:match_problems(output, matcher) abort
+  if has_key(a:matcher, 'pattern')
+    let items = []
+    for line in a:output
+      let rst = matchlist(line, a:matcher.pattern.regexp)
+      let file = rst[a:matcher.pattern.file]
+      let line = rst[a:matcher.pattern.line]
+      let column = rst[a:matcher.pattern.column]
+      let message = rst[a:matcher.pattern.message]
```

Comment:
1. Suggest using temp variable to hold matcher.pattern.*, so it may save some time for large output (Maybe tens of thousand lines for a large project).
2. Suggest using ``get()`` to access ``matcher.pattern`` to deal with some cases like lacking ``column`` field
3. Suggest using ``get()`` to access ``rst`` or judge whether the corrsponding fields exists. Because not all error lines contains all four file/line/column/message fields

Reply:
Ok. updated.